.github/workflows/pr-automation.yml

```
@@ -41,6 +50,23 @@ jobs:
           POOL_INSTANCE: ${{ steps.cloud_variables.outputs.POOL_INSTANCE }}
           BACKUP_ID: ${{ steps.cloud_variables.outputs.BACKUP_ID }}
 
+      - name: Generate shard matrix
+        id: generate-shard-matrix
+        env:
+          CONCURRENCY: ${{ inputs.shards || 4 }}
+        run: |
+          MAX_CONCURRENCY=4
+          if [ "$CONCURRENCY" -gt "$MAX_CONCURRENCY" ]; then
+            echo "[WARN] The requested number of shards ($CONCURRENCY) exceeds maximum allowed value ($MAX_CONCURRENCY). Using $MAX_CONCURRENCY shards instead." >&2
+            CONCURRENCY=$MAX_CONCURRENCY
+          fi
+          shard_matrix=()                             # Initialize an empty array to store the shard values.
+          for i in $(seq 1 "$CONCURRENCY"); do        # Loop through the numbers from 1 to $CONCURRENCY.
+            shard_matrix+=("\"$i/$CONCURRENCY\"")     # For each number i, append a string in the format "${i}/${CONCURRENCY}" to the shard_matrix array.
+          done
+          shard_matrix=$(IFS=,; echo "[${shard_matrix[*]}]") # Join the elements of the shard_matrix array with commas and wrap the result in square brackets to create a valid JSON array string.
+          echo "SHARD_MATRIX=$shard_matrix" >> "$GITHUB_OUTPUT"
```

We should be explicit that we only want the first item of the array otherwise it may be misleading:
```suggestion
          echo "SHARD_MATRIX=${shard_matrix[0]}" >> "$GITHUB_OUTPUT"
```